### PR TITLE
Support --master-data in mysql_db dumps

### PR DESCRIPTION
**Pull request by jonathonwalz**
##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

mysql_db
##### ANSIBLE VERSION

```
2.1.1.0
```
##### SUMMARY

This adds the ability for a user to specify the `--master-data` switch when dumping a mysql database. This is a rebase of #363 and closes #362.

Note that I didn't place the `master_data` parameter last so as to keep this solely a feature addition. #3687 fixes the issue where `single_transaction` and `quick` are not correctly handled. Happy to include the fix here or rebase after #3687 is merged if that is the preferred method of change tracking.

### CODE REVIEW

**Comment by jonathonwalz** [comment]
Migrated from https://github.com/ansible/ansible-modules-core/pull/4691

**Comment by Jmainguy** [comment]
I have not worked with mysql replication at all pretty much, can you explain the benefit of generating a commented 'CHANGE MASTER TO' by passing the value 2 to this?

**Comment by jonathonwalz** [comment]
I have no idea what the benefit is. It's just a standard option provided by the dump command.

**Comment by jonathonwalz** [comment]
@Jmainguy Are there any issues with this currently that I can address to get this merged?

**Comment by jonathonwalz** [comment]
@Jmainguy Is there anything I should be doing to get this merged?

**Comment by jctanner** [comment]
@jonathonwalz it would help if you added an integration test to assert the behavior does not regress over time.

https://github.com/ansible/ansible/tree/devel/test/integration/targets/mysql_db/tasks

**Comment by pnedkov** [comment]
I need this option in Ansible. I started digging and found this PR. Here is my workflow:

1. Dump the DB from Master with `--master-data=2`in order to have the commented `CHANGE MASTER TO...` line in the sql file

2. Import into Slave

3. Parse the dump file in order to find the values for MASTER_LOG_FILE and MASTER_LOG_POS

4. Configure the Slave with `CHANGE MASTER TO...` command where I have to provide the values for MASTER_LOG_FILE and MASTER_LOG_POS

5. Start slave

I know I can obtain the MASTER_LOG_FILE and MASTER_LOG_POS values from the Master with `show master status`. But since the position constantly changes I must know the position at the time when the dump has been taken.

Please merge. Thanks!

**Comment by jonathonwalz** [comment]
@pnedkov I probably won't have time to write tests for this for a while, it should theoretically speed up the merge process if tests were written. I've given you access to my fork if you have the time to write some.

**Comment by jctanner** [comment]
!waffling needs_revision

File: lib/ansible/modules/database/mysql/mysql_db.py [hunk 1]
```diff
@@ -56,6 +56,13 @@
       - Encoding mode to use, examples include C(utf8) or C(latin1_swedish_ci)
     required: false
     default: null
+  master_data:
+    description:
+      - Produce a dump from a master for seeding a slave. (1 to generate a 'CHANGE MASTER TO' statement;
+        2 to generate a commented 'CHANGE MASTER TO'; 0 or null to not include master data)
+    required: false
+    default: null
+    version_added: "2.4"
```
Comment (bmalynovytch) [inline lib/ansible/modules/database/mysql/mysql_db.py:65]: `version_added` should be 2.7 for now

**Review by bmalynovytch — CHANGES_REQUESTED** [review]

**Comment by Akasurde** [comment]
cc @pratikgadiya12

File: lib/ansible/modules/database/mysql/mysql_db.py (hunk 1 above)
Comment (markuman) [reply lib/ansible/modules/database/mysql/mysql_db.py:65]: ....2.10

**Review by markuman — CHANGES_REQUESTED** [review]
Okey, this feature is a must have (imho).

sadly the feature request  is very outdated and `lib/ansible/modules/database/mysql/mysql_db.py`  received a lot of updates.  
one of the biggest differences for python3 is, that `mysql_db` requires PyMySQL, but here it still requires `MySQLdb`.

Afaik, these commits are missing

```
30c2d21f17 mysql: add elements for list params to the modules' documentation (#63187)
393e4a41d0 Added support to create and delete multiple databases in MySQL (#58602)
44058e9425 Allow multiple databases(not all) to be dumped from mysql (#56721)
3b9478ade0 Migrate away from pipes.quote (#56785)
bc4ef99533 standardize TLS connection properties (#54315)
677c04c01d Sanity fixes to parameter types (#52325)
48642dd1d7 Copy/Paste error for ssl_ca (#51614)
73ffe683b2 Update documentation for mysql_db notes (#50444)
d34cf93f1a Migrate from MySQLdb to PyMySQL (#40123)
8d6522c671 Update mysql_db.py (#41317)
4ae4e80cc3 Add support for mysqldump --ignore-table switch (#33065)
cdd21e2170 Clean up module documentation (#36909)
9735a70059 Fix invalid string escape sequences.
f203ca7907 Update metadata to 1.1
dbf9a65c05 Improve error messages when MySQLdb is not installed (#27871)
0b9a78f0b3 Remove wildcard, add boilerplate and get rid of get_exception
```

maybe the easiest thing is simply to drop this request and start from current `devel` branch?   
because without `d34cf93f1a Migrate from MySQLdb to PyMySQL (#40123)`, it is currently not testable.


what do you think @bmalynovytch?  
Are you still interested  @jonathonwalz?  
cc @pratikgadiya12 @Akasurde

**Comment by Andersson007** [comment]
close_me

**Comment by Andersson007** [comment]
closed via https://github.com/ansible/ansible/pull/66048

**Pull request closed without merging**